packages/safe-ds-vscode/src/extension/eda/apis/runnerApi.ts

```diff
@@ -0,0 +1,415 @@
+                    newIDnessPlaceholderName,
+                );
+
+                // Find unique values
+                let uniqueValues = new Set();
+                for (let j = 0; j < columns[i]![1].values.length; j++) {
+                    uniqueValues.add(columns[i]![1].values[j]);
+                }
+                if (uniqueValues.size <= 3 || uniqueValues.size > 10) {
+                    // Must match conidtions below that choose to display histogram
+                    continue; // This historam only generated if between 4-10 categorigal uniques or numerical type
+                }
+            }
+
+            // Histogram for numerical columns or categorical columns with 4-10 unique values
+            const newHistogramPlaceholderName = this.genPlaceholderName();
+            columnNameToPlaceholderHistogramNameMap.set(columns[i]![1].name, newHistogramPlaceholderName);
+            sdsStrings += this.sdsStringForHistogramByColumnName(
+                columns[i]![1].name,
+                table.name,
+                newHistogramPlaceholderName,
+            );
+        }
+
+        // Execute with generated SDS code
+        const pipelineExecutionId = crypto.randomUUID();
+        await this.addToAndExecutePipeline(pipelineExecutionId, sdsStrings);
+
+        // Get missing value ratio for each column
+        for (const [, placeholderName] of columnNameToPlaceholderMVNameMap) {
+            const missingValueRatio = await this.getPlaceholderValue(placeholderName, pipelineExecutionId);
+            if (missingValueRatio) {
+                missingValueRatioMap.set(placeholderName, missingValueRatio as number);
+            }
+        }
+
+        // Get IDness for each column
+        for (const [, placeholderName] of columnNameToPlaceholderIDnessNameMap) {
+            const idness = await this.getPlaceholderValue(placeholderName, pipelineExecutionId);
+            if (idness) {
+                idnessMap.set(placeholderName, idness as number);
+            }
+        }
+
+        // Get histogram for each column
+        for (const [, placeholderName] of columnNameToPlaceholderHistogramNameMap) {
+            const histogram = await this.getPlaceholderValue(placeholderName, pipelineExecutionId);
+            if (histogram) {
+                histogramMap.set(placeholderName, histogram as Base64Image);
+            }
+        }
+
+        // Create profiling data
+        const profiling: { columnName: string; profiling: Profiling }[] = [];
+        for (const column of columns) {
+            // Base info for the top of the profiling
+            const missingValuesRatio =
+                missingValueRatioMap.get(columnNameToPlaceholderMVNameMap.get(column[1].name)!)! * 100;
+
+            const validRatio: ProfilingDetailStatistical = {
+                type: 'numerical',
+                name: 'Valid',
+                value: missingValuesRatio ? (100 - missingValuesRatio).toFixed(2) + '%' : '100%',
```

Comment:
Keep the ratio here. Do the conversion to a string (representation) in the Svelte component. Now that the profiling data is no longer only separated into top & bottom, this is possible.